Content.Server/Atmos/EntitySystems/DeltaPressureSystem.cs

```diff
@@ -0,0 +1,283 @@
+    {
+        var aboveMinPressure = pressure > ent.Comp.MinPressure;
+        var aboveMinDeltaPressure = deltaPressure > ent.Comp.MinPressureDelta;
+        if (!aboveMinPressure && !aboveMinDeltaPressure)
+        {
+            ent.Comp.IsTakingDamage = false;
+            return;
+        }
+
+        // shitcode
+        var baseDamage = ent.Comp.BaseDamage;
+        var appliedDamage = ent.Comp.BaseDamage;
+        if (aboveMinPressure)
+        {
+            appliedDamage = MutateDamage(ent, baseDamage, pressure - ent.Comp.MinPressure);
+        }
+        if (aboveMinDeltaPressure)
+        {
+            if (ent.Comp.StackDamage)
+            {
+                appliedDamage += MutateDamage(ent, appliedDamage, deltaPressure - ent.Comp.MinPressureDelta);
+            }
+            else
+            {
+                appliedDamage = MutateDamage(ent, baseDamage, deltaPressure - ent.Comp.MinPressureDelta);
+            }
+        }
+
+        _damage.TryChangeDamage(ent, appliedDamage, ignoreResistances: true, interruptsDoAfters: false);
+        ent.Comp.IsTakingDamage = true;
+    }
+
+    /// <summary>
+    /// Mutates the damage dealt by a DamageSpecifier based on values on an entity with a DeltaPressureComponent.
+    /// </summary>
+    /// <param name="ent">The entity to base the manipulations off of (pull scaling type)</param>
+    /// <param name="damage">The damage specifier to mutate.</param>
+    /// <param name="pressure">The pressure being exerted on the entity.</param>
+    /// <returns>The mutated DamageSpecifier.</returns>
+    [PublicAPI]
+    public static DamageSpecifier MutateDamage(Entity<DeltaPressureComponent> ent, DamageSpecifier damage, float pressure)
+    {
+        switch (ent.Comp.ScalingType)
+        {
+            case DeltaPressureDamageScalingType.Threshold:
+                break;
+
+            case DeltaPressureDamageScalingType.Linear:
+                damage *= pressure * ent.Comp.ScalingPower;
+                break;
+
+            case DeltaPressureDamageScalingType.Log:
+                // This little line's gonna cost us 51 CPU cycles
+                damage *= Math.Log(pressure, ent.Comp.ScalingPower);
+                break;
+
+            case DeltaPressureDamageScalingType.Exponential:
+                damage *= Math.Pow(pressure, ent.Comp.ScalingPower);
+                break;
+            default:
+                throw new ArgumentOutOfRangeException(nameof(ent), "Invalid damage scaling type!");
+        }
+
+        return damage;
+    }
+
+    /// <summary>
+    /// Sets the base damage applied to the entity per atmos tick when above the damage threshold.
+    /// </summary>
+    /// <param name="ent">The entity whose component to modify.</param>
+    /// <param name="baseDamage">The new base damage specifier.</param>
+    [PublicAPI]
+    public static void SetBaseDamage(Entity<DeltaPressureComponent> ent, DamageSpecifier baseDamage)
```

Comment:
Honestly I think just tag these fields you have setters for with `[Access(Other = AccessPermissions.ReadWrite)]`. Unless there's some reason in the future that DeltaPressureSystem would need to have custom setter logic, I feel like it's fine. It's a pretty niche breaking change to switch back to setters anyway if that does end up happening.